### PR TITLE
Turret control

### PR DESCRIPTION
**Pull request by ClaudP**
Pull Request Checklist:
- [x] You are able to understand what the code is doing on a basic level with the use of comments
- [x] Code is formatted in a way that isn't confusing
- [x] builds cleanly
- [x] builds *to the robot* cleanly
- [ ] Code has been tested on the robot/roadkill/other mechanical setup and completes the task it needs to complete

### CODE REVIEW

File: src/main/java/frc/robot/components/InterpolatorTable.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.components;
+
+/*
+* Add your docs here.
```
Comment (122004) [inline src/main/java/frc/robot/components/InterpolatorTable.java]: We should add our docs here

File: src/main/java/frc/robot/components/InterpolatorTable.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        int maxIdx = -1;
+        double minDistance = 1000000000;
+        double maxDistance = -1000000000;
+        for (int i = 0; i < entries.length; i++) {
+            double value = entries[i].x;
+            double distance = input - value;
+            if (distance < 0 && Math.abs(distance) < Math.abs(minDistance)) {
+                minIdx = i;
+                minDistance = distance;
+            } else if (distance >= 0 && Math.abs(distance) < Math.abs(maxDistance)) {
```
Comment (122004) [inline src/main/java/frc/robot/components/InterpolatorTable.java:34]: Mas comentarios por favor.

File: src/main/java/frc/robot/components/InterpolatorTable.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                maxDistance = distance;
+            }
+        }
+        if (minIdx == -1) {
+            return entries[maxIdx].y;
+        } else if (maxIdx == -1) {
+            return entries[minIdx].y;
+        }
+        double interpolationValue = (input - entries[minIdx].x) / (entries[maxIdx].x - entries[minIdx].x);
+        return entries[minIdx].y + (entries[maxIdx].y - entries[minIdx].y) * interpolationValue;
```
Comment (122004) [inline src/main/java/frc/robot/components/InterpolatorTable.java:45]: Mehr kommentare.

**Review by Linkd-Aether — DISMISSED** [review]

File: src/main/java/frc/robot/subsystems/instances/Magazine.java [hunk 4]
```diff
@@ -45,48 +48,31 @@ public void periodic() {
...
+   * if a ball passes through the exit, a power cell is decreased
+   */
   public void updateCellCount() {
 
-    if (entrance.get() == true && cellEntering == false && cellCount < 6) {
+    if(enterance.get() == true){
       cellCount++;
-      cellEntering = true;
     }
-
```
Comment (122004) [inline src/main/java/frc/robot/subsystems/instances/Magazine.java]: So uh this is wrong please don't delete my stuff

**Review by 122004 — CHANGES_REQUESTED** [review]
There are several things wrong with this branch. Please DO NOT merge until I fix the issues.

**Review by Linkd-Aether — DISMISSED** [review]

File: src/main/java/frc/robot/components/InterpolatorTable.java (hunk 1 above)
Comment (122004) [reply src/main/java/frc/robot/components/InterpolatorTable.java]: I'll just do it

**Review by Linkd-Aether — DISMISSED** [review]

File: src/main/java/frc/robot/components/InterpolatorTable.java (hunk 2 above)
Comment (BenWertz) [reply src/main/java/frc/robot/components/InterpolatorTable.java:34]: Te entiendo, y lo seré en mi proximo cometido. Pues, si quieres hacerlo entonces puedo find una otra cosa. Todo de que hablo es que _una_ persona necesita que hacerlo, y si no hay otras personas que tienen tiempo libre, yo me pongo en el trabajo.

**Review by BenWertz — DISMISSED** [review]
Hopefully everything will be okay merge-wise.

File: src/main/java/frc/robot/components/InterpolatorTable.java (hunk 1 above)
Comment (122004) [reply src/main/java/frc/robot/components/InterpolatorTable.java]: It has been done.

File: src/main/java/frc/robot/subsystems/instances/Magazine.java (hunk 4 above)
Comment (122004) [reply src/main/java/frc/robot/subsystems/instances/Magazine.java]: It has been fixed.

**Review by Linkd-Aether — APPROVED** [review]

**Review by BenWertz — APPROVED** [review]

**Review by jjesse1972 — APPROVED** [review]

**Pull request merged**